AK/Format.h

```diff
@@ -56,95 +56,157 @@ concept Formattable = HasFormatter<T>;
-            return get_type_from_size<sizeof(T), IsUnsigned<T>>();
-        else
-            return Type::Custom;
     }
 
     template<typename Visitor>
     constexpr auto visit(Visitor&& visitor) const
     {
         switch (type) {
-        case TypeErasedParameter::Type::UInt8:
-            return visitor(*static_cast<u8 const*>(value));
-        case TypeErasedParameter::Type::UInt16:
-            return visitor(*static_cast<u16 const*>(value));
-        case TypeErasedParameter::Type::UInt32:
-            return visitor(*static_cast<u32 const*>(value));
-        case TypeErasedParameter::Type::UInt64:
-            return visitor(*static_cast<u64 const*>(value));
-        case TypeErasedParameter::Type::Int8:
-            return visitor(*static_cast<i8 const*>(value));
-        case TypeErasedParameter::Type::Int16:
-            return visitor(*static_cast<i16 const*>(value));
-        case TypeErasedParameter::Type::Int32:
-            return visitor(*static_cast<i32 const*>(value));
-        case TypeErasedParameter::Type::Int64:
-            return visitor(*static_cast<i64 const*>(value));
-        default:
-            TODO();
+        case Type::UnsignedInteger:
+            return visitor(value.as_unsigned);
+        case Type::SignedInteger:
+            return visitor(value.as_signed);
+        case Type::Boolean:
+            return visitor(value.as_bool);
+        case Type::Character:
+            return visitor(value.as_char);
+#ifndef KERNEL
+        case Type::Float:
+            return visitor(value.as_float);
+        case Type::Double:
+            return visitor(value.as_double);
+#else
+        case Type::Float:
+        case Type::Double:
+            // Floating-point types are not allowed in kernel mode.
+            VERIFY_NOT_REACHED();
+#endif
+        case Type::StringView:
+            return visitor(value.as_string_view);
+        case Type::CString:
+            return visitor(value.as_c_string);
+        case Type::CustomType:
+            return visitor(value.as_custom_type);
         }
+        VERIFY_NOT_REACHED();
     }
 
     constexpr size_t to_size() const
     {
-        return visit([]<typename T>(T value) {
-            if constexpr (sizeof(T) > sizeof(size_t))
-                VERIFY(value < NumericLimits<size_t>::max());
-            if constexpr (IsSigned<T>)
+        return visit([]<typename T>(T value) -> size_t {
+            if constexpr (IsSame<T, u64>)
+                return static_cast<size_t>(value);
+
+            if constexpr (IsSame<T, i64>) {
                 VERIFY(value >= 0);
-            return static_cast<size_t>(value);
+                return static_cast<size_t>(value);
+            }
+
+            TODO();
         });
     }
 
-    // FIXME: Getters and setters.
-
-    void const* value;
+    union {
+        u64 as_unsigned;
+        i64 as_signed;
+        bool as_bool;
+        char as_char;
+        float as_float;
+        double as_double;
```

Comment:
#ifdef KERNEL required?

Reply:
Seemed to build without, as storing them seems to be fine,
only needed to add the `ifndef`s in the invocation part

So should ifdef them here, and should I also ifdef out the enum values?
(also maybe GCC CI has a different opinion than my local clang)